pkg/storage/zip.go

```diff
@@ -0,0 +1,9 @@
+package storage
+
+import "io"
+
+// Zip represents zip file of a specific version and it's size
+type Zip struct {
+	Zip  io.ReadCloser
```

Comment:
@manugupt1 I noticed in code that uses this, there are a lot of references to `.Zip.Zip`. This is definitely a nit-pick, but could you rename it so it's not redundant? Maybe it could be `Reader`?

Reply:
Also what do you think of embeding the interface?
```
type Zip struct {
    io.ReadCloser
    Size int64
}
```

Reply:
I am okay with it, but can we do it in a different PR. This PR does too many things imo.